x-pack/legacy/plugins/reporting/server/usage/reporting_usage_collector.ts

```diff
@@ -57,23 +57,35 @@ export function getReportingUsageCollector(
 
 export function registerReportingUsageCollector(
   reporting: ReportingCore,
-  plugins: ReportingSetupDeps
-) {
-  if (!plugins.usageCollection) {
+  { licensing, usageCollection }: ReportingSetupDeps
+): void {
+  if (!usageCollection) {
     return;
   }
-  const xpackMainInfo = plugins.__LEGACY.plugins.xpack_main.info;
 
+  const config = reporting.getConfig();
   const exportTypesRegistry = reporting.getExportTypesRegistry();
+  const getLicense = async () => {
+    return await licensing.license$
+      .pipe(
+        map(({ isAvailable, type }) => ({
```

Comment:
The `FeaturesAvailable` type being implemented here carries over the shape of the data from the Legacy xpack_main. It avoid the underlying code needing extra changes.